src/core/dbus-manager.c

```diff
@@ -48,6 +50,10 @@
 #include "virt.h"
 #include "watchdog.h"
 
+/* Require 16MiB free in /run/systemd for reloading/reexecing. After all we need to serialize our state there, and if
+ * we can't we'll fail badly. */
+#define RELOAD_DISK_SPACE_MIN (UINT64_C(16) * UINT64_C(1024) * UINT64_C(1024))
```

Comment:
Actually, they aren't casts, and yes, they do make a difference here...

They do two things: they suffix the numbers with "U" and with "LL". The "U" has the effect that the numeric constant is unsigned. That's a good thing because otherwise gcc with the fascist compiler options we use would complain about comparing an unsigned with a signed if we compare the value with disk sizes (and disk sizes are usually unsigned). The "LL" actually makes it possible to use large values above 32bit here... If we don't, then the compiler might use something smaller here at its discretion (and in fact, gcc will)...

The suffixes are actually weird, because unlike the name suggests UINT64_C doesn't actually result in something of type uint64_t. It just means that it adds a suffix sufficient to make constant compatible with that type...

We could also write this as 16U*1024U*1024U btw. On systems where "int"s are 32bit, that's sufficient. However, I actually want to clarify here that disk sizes are stored in uint64_t, and hence I am a bit more explicit here. Also, while developing this i changed this for testing purposes to 16U*1024U*1024U*1024U (i.e. 16GiB instea dof 16MiB, to ensure the limit is hit on my testin machine) and in that case, if you just use U then things blow up in your face as the resulting product cannot be stored in a 32bit integer anymore... Hence you have to use ULL int that case, and that's precisely what UINT64_C resolves to, but in a way that doesn't encode the lengths of a long or long long...

Yeah, C is weird...